### PR TITLE
Refactors google calendar service

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Uses zod to actually validate API keys and user Credentials
- Optimizes auth legibility
- Uses fetcher helper to simplify API calls

## Type of change

- [x] Chore

## How should this be tested?

- [ ] Test Google Calendar integration

### CODE REVIEW

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -326,22 +303,20 @@ export default class GoogleCalendarService implements Calendar {
               },
             },
             (err, apires) => {
-              if (err) {
+              if (err || !apires?.data.calendars) {
```
Comment (zlwaterfield) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:306]: `!apires?.data.calendars` doesn't mean `err` will be defined, would it be better to have another check?

```
if (!apires?.data.calendars) {
	reject("No calendar data found...")
	return;
}

**Review by zlwaterfield — COMMENTED** [review]
@zomars what is the best way to test this locally? I don't see Google Calendar as an app locally.

File: packages/app-store/googlecalendar/lib/CalendarService.ts (hunk 1 above)
Comment (zomars) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:306]: That check passes if any of those is true

Comment (zomars) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:306]: Here are the instructions to [setup Google](https://github.com/calcom/cal.com#obtaining-the-google-api-credentials)

Comment (zlwaterfield) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:306]: Yes but if `err` is `null` and it hits on `!apires?.data.calendars` then you'll be rejecting on null (`reject(null)`) and there will be no context in why the error occurred.

Comment (zomars) [reply packages/app-store/googlecalendar/lib/CalendarService.ts:306]: Agreed. will update

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**